plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -7967,13 +7967,53 @@ public void testNonSelectStatementsWithPartitionFilterRequired()
         assertUpdate(session, "DROP TABLE " + tableName2);
     }
 
+    @Test
+    public void testPartitionFilterRequiredSchemas()
+    {
+        String schemaName = "gold_mart";
+        String tableName = "test_partition_" + randomNameSuffix();
+
+        Session session = withPartitionFilterRequiredSchemas(withPartitionFilterRequired(getSession()));
```

Comment:
Little hard to imagine the generated session properties from this nested method call. I would recommend inlining the builder:
```java
        Session session = Session.builder(getSession())
                .setCatalogSessionProperty("iceberg", "query_partition_filter_required", "true")
                .setCatalogSessionProperty("iceberg", "query_partition_filter_required_schemas", "[\"tpch\"]")
                .build();
```